app/src/main/java/com/hamdan/forzenbook/view/login/MainLogin.kt

```diff
@@ -45,70 +51,95 @@ fun MainLoginContent(
+    onTextChange: (Entry, Entry, Boolean) -> Unit,
     onSubmission: () -> Unit
 ) {
-    var email: String = stateEmail
-    var emailError = stateEmailError
+    var email = stateEmail.text
+    var emailError = stateEmail.error
+    var code = stateCode.text
+    var codeError = stateCode.error
     val focusManager = LocalFocusManager.current
     val keyboardController = LocalSoftwareKeyboardController.current
-    InputField(
-        label = stringResource(R.string.login_email_prompt), value = email,
-        onValueChange = {
-            email = it
-            emailError =
-                ((email.length > 30) || (!TextUtils.isEmpty(email) && !validateEmail(email)))
-            onTextChange(email, emailError)
-        },
-        imeAction = ImeAction.Next, onNext = { focusManager.moveFocus(FocusDirection.Down) }
-    )
-    if (emailError && email != "") {
-        Text(text = stringResource(R.string.login_email_error))
+
+    if (isInputtingCode) {
+        InputField(
+            label = stringResource(R.string.login_email_prompt), value = email,
+            onValueChange = {
+                email = it
+                emailError = email.length > EMAIL_LENGTH_LIMIT || !validateEmail(email)
+                onTextChange(Entry(email, emailError), Entry(code, codeError), isInputtingCode)
+            },
+            imeAction = ImeAction.Next, onNext = { focusManager.moveFocus(FocusDirection.Down) }
+        )
+        if (emailError && email != "") {
+            Text(text = stringResource(R.string.login_email_error))
+        }
+        InputField(
+            label = stringResource(R.string.login_code_prompt), value = code,
+            onValueChange = {
+                code = it
+                codeError =
+                    (code.length > CODE_LENGTH_MAX)
+                onTextChange(Entry(email, emailError), Entry(code, codeError), isInputtingCode)
+            },
+            keyboardType = KeyboardType.Number,
+            imeAction = ImeAction.Done,
+            onDone = {
+                keyboardController?.hide()
+                if (!emailError || !codeError) {
+                    onSubmission()
+                }
+            },
+        )
+        if (codeError && code != "") {
+            Text(text = stringResource(R.string.login_code_error))
+        }
+    } else {
+        InputField(
+            label = stringResource(R.string.login_email_prompt), value = email,
+            onValueChange = {
+                email = it
+                emailError = email.length > EMAIL_LENGTH_LIMIT || !validateEmail(email)
+                onTextChange(Entry(email, emailError), Entry(code, codeError), isInputtingCode)
+            },
+            imeAction = ImeAction.Done,
+            onDone = {
+                keyboardController?.hide()
+                if (!emailError) {
+                    onSubmission()
+                }
+            },
+        )
+        if (emailError && email != "") {
```

Comment:
You can use functions like isEmpty or isNotEmpty instead of checking against nothing

Reply:
updated all instances to be isNotEmpty()